msbuild/Xamarin.Mac.Tasks.Core/Tasks/ArchiveTaskBase.cs

```diff
@@ -24,10 +28,22 @@ public override bool Execute ()
 				var appDestDir = Path.Combine (productsDir, "Applications", Path.GetFileName (AppBundleDir.ItemSpec));
 				Ditto (AppBundleDir.ItemSpec, appDestDir);
 
-				// Archive the dSYMs...
-				if (Directory.Exists (DSYMDir)) {
-					var destDir = Path.Combine (archiveDir, "dSYMs", Path.GetFileName (DSYMDir));
-					Ditto (DSYMDir, destDir);
+				// Archive the main dSYM...
```

Comment:
Would it be complicated to merge the iOS and Mac versions of ArchiveTaskBased?

We'll most likely need this logic for Mac Catalyst, but we're using the iOS version of the task assemblies...

Reply:
> Would it be complicated...

hmmm... a definite nyes :) 
It would not really be complicated but it would (should) not be a merge like it could be done for most other tasks and best done in a different PR, since the current code is not easy to share.

From memory (likely incomplete) this means
* deal with different paths for `Frameworks`, e.g. extra `Contents` on macOS
* deal with different paths for extensions, e.g. `PlugIns` for macOS. Not sure for Catalyst - is it even possible to have iOS extensions in them ?!?
* AOT differences, e.g. bunch of `.dSYM` for macOS
* Native libraries, e.g. deal with `MonoBundle` on macOS, but not on other platforms
* Bitcode handling - not applicable on macOS

I think the simplest approach would be to have the (shared) base class include a bunch of helper methods (I added `ArchiveDSym` in this PR) and make very simple (and likely not shared) platform specific tasks.